plugins/database/hana/hana.go

```diff
@@ -0,0 +1,284 @@
+	// Grab the lock
+	h.Lock()
+	defer h.Unlock()
+
+	// Get the connection
+	db, err := h.getConnection()
+	if err != nil {
+		return "", "", err
+	}
+
+	if statements.CreationStatements == "" {
+		return "", "", dbutil.ErrEmptyCreationStatement
+	}
+
+	// Generate username
+	username, err = h.GenerateUsername(usernamePrefix)
+	if err != nil {
+		return "", "", err
+	}
+
+	// HANA does not allow hyphens in usernames, and highly prefers capital letters
+	username = strings.Replace(username, "-", "_", -1)
+	username = strings.ToUpper(username)
+
+	// Generate password
+	password, err = h.GeneratePassword()
+	if err != nil {
+		return "", "", err
+	}
+
+	// Most HANA configurations have password constraints.
+	// Prefix with A1a (user must change password upon login anyway)
+	password = strings.Replace(password, "-", "_", -1)
+	password = "A1a" + password
+
+	// expirationStr = (expiration - time.Now).Seconds() + HANA server side time
+	// This ensures hana will deactivate the user when time is up, regardless of vault's actions
+	var expirationStr string
+	timeQuery := fmt.Sprintf("SELECT TO_NVARCHAR(add_seconds(CURRENT_TIMESTAMP,"+
+		"%f), 'YYYY-MM-DD HH24:MI:SS') FROM DUMMY", (expiration.Sub(time.Now())).Seconds())
+	if err := db.QueryRow(timeQuery).Scan(&expirationStr); err != nil {
+		return "", "", err
+	}
+
+	// Start a transaction
+	tx, err := db.Begin()
+	if err != nil {
+		return "", "", err
+	}
+	defer tx.Rollback()
+
+	// Execute each query
+	for _, query := range strutil.ParseArbitraryStringSlice(statements.CreationStatements, ";") {
+		query = strings.TrimSpace(query)
+		if len(query) == 0 {
+			continue
+		}
+
+		stmt, err := tx.Prepare(dbutil.QueryHelper(query, map[string]string{
+			"name":       username,
+			"password":   password,
+			"expiration": expirationStr,
+		}))
+		if err != nil {
+			return "", "", err
+		}
+		defer stmt.Close()
+		if _, err := stmt.Exec(); err != nil {
+			return "", "", err
+		}
+	}
+
+	// Commit the transaction
+	if err := tx.Commit(); err != nil {
+		return "", "", err
+	}
+
+	return username, password, nil
+}
+
+// Renewing hana user just means altering user's valid until property
+func (h *HANA) RenewUser(statements dbplugin.Statements, username string, expiration time.Time) error {
+	// Get connection
+	db, err := h.getConnection()
+	if err != nil {
+		return err
+	}
+
+	// Start a transaction
+	tx, err := db.Begin()
+	if err != nil {
+		return err
+	}
+	defer tx.Rollback()
+
+	// Request server's current time plus lease duration
+	var expirationStr string
+	timeQuery := fmt.Sprintf("SELECT TO_NVARCHAR(add_seconds(CURRENT_TIMESTAMP,"+
+		"%f), 'YYYY-MM-DD HH24:MI:SS') FROM DUMMY", (expiration.Sub(time.Now())).Seconds())
+	if err := db.QueryRow(timeQuery).Scan(&expirationStr); err != nil {
+		return err
+	}
+
+	// Renew user's valid until property field
+	stmt, err := tx.Prepare("ALTER USER " + username + " VALID UNTIL " + "'" + expirationStr + "'")
```

Comment:
If you are asking the use case for this particular SQL:

If `expiration` field is in the role's SQL, the generated credential will be set to expire when the lease expires on HANA server side. This means, in order to renew a credential, the expiration date that HANA keeps for that user must also be pushed back. Hence this `ALTER USER` sql.

If you are asking for the use case of potential custom renew statements that users may want to input:

I am unaware of any workflows where this would be needed. But if the convention is to make custom renew statements available, I can do that too.